src/OrchardCore.Modules/OrchardCore.ContentTypes/Services/ContentDefinitionService.cs

```diff
@@ -132,7 +141,7 @@ public async Task<ContentTypeDefinition> AddTypeAsync(string name, string displa
             ContentTypeDefinition = contentTypeDefinition,
         };
 
-        _contentDefinitionEventHandlers.Invoke((handler, ctx) => handler.ContentTypeCreated(ctx), context, _logger);
+        TriggerContentTypeCreated(context);
```

Comment:
all these changes should not be needed. In fact, I don't think the service class should be modified from the original class.